aidial_adapter_openai/utils/tokenizer.py

```diff
@@ -0,0 +1,125 @@
+"""
+Implemented based on the official recipe: https://cookbook.openai.com/examples/how_to_count_tokens_with_tiktoken
+"""
+
+from abc import abstractmethod
+from typing import Generic, List, TypeVar
+
+from aidial_sdk.exceptions import InternalServerError
+from tiktoken import Encoding, encoding_for_model
+
+from aidial_adapter_openai.utils.image_tokenizer import tokenize_image_by_size
+from aidial_adapter_openai.utils.multi_modal_message import MultiModalMessage
+
+TokenizeInput = TypeVar("TokenizeInput")
```

Comment:
The tokenizer input is a rather a list of these thing, not this thing itself.
Why not `MessageType`?

Reply:
Fixed